src/Message/ApplePayAuthorizeResponse.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Omnipay\Vindicia\Message;
+
+class ApplePayAuthorizeResponse extends Response
+{
+    /**
+     * Is the response successful?
+     * Throws an exception if there's no code.
+     *
+     * @return boolean
+     * @throws \Omnipay\Common\Exception\InvalidResponseException
+     */
+    public function isSuccessful()
+    {
+        $statusCode = $this->getStatusCode();
+        return ($statusCode >= 200 && $statusCode < 300) || $statusCode == 304;
+    }
+
+    /**
+     * Is the response successful?
+     * Throws an exception if there's no code.
+     *
+     * @return boolean
+     * @throws \Omnipay\Common\Exception\InvalidResponseException
+     */
+    public function getReason()
+    {
+        return $this->data['reason'];
+    }
+
+    /**
+     * Is the response successful?
+     * Throws an exception if there's no code.
+     *
+     * @return boolean
+     * @throws \Omnipay\Common\Exception\InvalidResponseException
+     */
+    public function getStatusCode()
+    {
+        return $this->data['statusCode'];
+    }
+
+    /**
+     * Is the response a redirect?
+     * Throws an exception if there's no code.
+     * Successful PayPal purchase responses are always redirects.
+     *
+     * @return boolean
+     * @throws \Omnipay\Common\Exception\InvalidResponseException
+     */
+    public function isRedirect()
+    {
+        return $this->isSuccessful();
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getRedirectUrl()
+    {
+        $transaction = $this->getTransaction();
+        if (isset($transaction)) {
+            return $transaction->getPayPalRedirectUrl();
+        }
+        return null;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getApplePaySessionTimeStamp()
+    {
+        if (isset($this->data['epochTimestamp'])) {
+            return $this->data['epochTimestamp'];
+        }
+
+        return null;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getApplePaySessionExpirationTimeStamp()
+    {
+        if (isset($this->data['expiresAt'])) {
+            return $this->data['expiresAt'];
+        }
+
+        return null;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getApplePaySessionMerchantSessionID()
+    {
+        if (isset($this->data['merchantSessionIdentifier'])) {
+            return $this->data['merchantSessionIdentifier'];
+        }
+
+        return null;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getApplePaySessionNonceToken()
```

Comment:
can you add docs explaining what all these apple pay specific things are?

Reply:
Sure, they are all what you get back from Apple. I'll add a comment.

Reply:
Since I'm returning the Apple Pay Payment Session object as one json object, I no longer think these individual pieces of it are necessary. Apple knows how to handle expired session objects anyway.

Reply:
perfect!